sklearn/model_selection/_search.py

```diff
@@ -997,3 +1004,402 @@ def fit(self, X, y=None, labels=None):
+        cat_dict = {}
+        for param, param_vals in self.param_distributions.items():
+            if (np.iterable(param_vals) and
+                isinstance(param_vals[0], six.string_types)):
+                    le = LabelEncoder()
+                    le.fit(param_vals)
+                    cat_dict[param] = le
+
+        cv_scores, grid_scores, tested_parameters = [], [], []
+        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
+        cv = check_cv(self.cv, y, classifier=is_classifier(self.estimator))
+
+        #  Initialize with random candidates
+        init_candidates = ParameterSampler(
+            self.param_distributions, self.n_init,
+            random_state=self.random_state)
+
+        tested_parameters, cv_scores, grid_scores = self._evaluate_params(
+            init_candidates, cv, X, y, labels, cat_dict)
+
+        rng = check_random_state(self.random_state)
+        gp_params = self.gp_params
+        if not gp_params:
+            length_scale = np.ones(len(tested_parameters[0]))
+            gp_params = {'kernel': RBF(length_scale=length_scale)}
+
+        for i in range(self.n_iter - self.n_init):
+            # Model with a Gaussian Process
+            gp = GaussianProcessRegressor(**self.gp_params)
+            gp.fit(tested_parameters, cv_scores)
+
+            # Sample candidates and predict their corresponding
+            # acquisition values
+            candidates = ParameterSampler(
+                self.param_distributions, self.n_candidates,
+                random_state=rng)
+            candidate_values = []
+            candidate_dict = []
+            for s in candidates:
+                # get the candidates as an array to fit in the GPRegressor.
+                encoded_values = []
+                for candidate_key, value in s.items():
+                    if candidate_key in cat_dict:
+                        encoded_values.append(
+                            cat_dict[candidate_key].transform([value])[0])
+                    else:
+                        encoded_values.append(value)
+                candidate_values.append(encoded_values)
+                candidate_dict.append(s)
+
+            if self.acquisition_function == 'UCB':
+                predictions, std = gp.predict(
+                    candidate_values, return_std=True)
+                acquisition_func = predictions + self.kappa * std
+            elif self.acquisition_function == 'EI':
+                predictions, std = gp.predict(
+                    candidate_values, return_std=True)
+                y_best = np.max(cv_scores)
+                Z = (predictions - y_best - self.xi) / std
+                acquisition_func = std * (
+                    Z * stats.norm.cdf(Z) + stats.norm.pdf(Z))
+            else:
+                raise ValueError(
+                    'acquisition_function not implemented yet : '
+                    + self.acquisition_function)
```

Comment:
This could be done if the input was scaled appropriately. For example, if the regularization parameter was log-scaled, but it isn't, and you don't want to optimize it when its not log-scaled. e.g.

log-scaled
![image](https://cloud.githubusercontent.com/assets/277639/12637359/c70a6032-c594-11e5-854f-973ef51f0e0e.png)

vs non log-scaled

![image](https://cloud.githubusercontent.com/assets/277639/12637364/da50fbc4-c594-11e5-9282-bf4bc7f6ea22.png)
so we restore to sampling from a distribution (hence the Parametersampler), the expon in this case, which simulates this transformation.